### PR TITLE
[WT18] Port parsers to windows

### PR DESCRIPTION
**Pull request by Sckorn**

### CODE REVIEW

File: parsers/BUILD [hunk 1]
```diff
@@ -156,6 +164,26 @@ cc_test(
     ],
 )
 
+cc_test(
+    name = "rdmn_gltf_parsers_test",
+    srcs = [
+        "test/parsers_test.cpp",
+        "test/gltf_parser_test.cpp",
+    ],
+    copts = ["-Iexternal/gtest/include"],
```
Comment (radameon) [inline parsers/BUILD:173]: Does it compile without this one?

File: parsers/src/x3d/x3d_parser.cpp [hunk 2]
```diff
@@ -75,7 +85,78 @@ void parse_tex_coord_index(std::string &&str, rcbe::geometry::Mesh::FacetStorage
...
 }
+#ifdef R_TINYXML
+std::vector<rcbe::geometry::Mesh::TriangleType> parse_facets(const tinyxml2::XMLElement *element) {
+    auto *cind_attr = element->Attribute("coordIndex");
+    if (!cind_attr)
+        return {};
+
+    auto facets = parse_facets(std::string(cind_attr));
+
+    auto* tex_coords_elem = element->ToElement()->Attribute("texCoordIndex");
```
Comment (radameon) [inline parsers/src/x3d/x3d_parser.cpp:96]: Asterisks with auto are redundant

File: parsers/src/x3d/x3d_parser.cpp [hunk 3]
```diff
@@ -75,7 +85,78 @@ void parse_tex_coord_index(std::string &&str, rcbe::geometry::Mesh::FacetStorage
...
+}
+
+rcbe::geometry::Mesh parse_shape(const tinyxml2::XMLElement *shape_node) {
+    auto *ifs_elem = shape_node->FirstChildElement("IndexedFaceSet")->ToElement();
+    if (!ifs_elem)
+        return {};
+        
+    auto facets = parse_facets(ifs_elem);
+    auto vertices = parse_vertices(ifs_elem);
+    auto normals = parse_normals(ifs_elem);
```
Comment (radameon) [inline parsers/src/x3d/x3d_parser.cpp:135]: I'd suggest throwing if any of these is empty, or at least return a defaulted `Mesh`.

File: parsers/src/x3d/x3d_parser.cpp [hunk 4]
```diff
@@ -99,24 +180,48 @@ rcbe::geometry::Mesh parse_shape(const pt::ptree &subtree) {
...
 std::vector<geometry::Mesh> parse_meshes(const std::string &file_path) {
     return parse_meshes(std::ifstream {file_path});
 }
 
 std::vector<geometry::Mesh> parse_meshes(std::istream &&input_stream) {
+    std::vector<geometry::Mesh> ret;
     pt::ptree pt;
     pt::read_xml(input_stream, pt);
 
-    std::vector<geometry::Mesh> ret;
```
Comment (radameon) [inline parsers/src/x3d/x3d_parser.cpp:113]: Why?

File: parsers/src/x3d/x3d_parser.cpp (hunk 3 above)
Comment (Sckorn) [reply parsers/src/x3d/x3d_parser.cpp:135]: There are no checks in original code as well, although I do believe that `boost:property_tree` throws, when we request a path which is not present in XML. So I'll throw here as well.

File: parsers/src/x3d/x3d_parser.cpp (hunk 4 above)
Comment (Sckorn) [reply parsers/src/x3d/x3d_parser.cpp:113]: Reverted

File: parsers/src/x3d/x3d_parser.cpp (hunk 3 above)
Comment (Sckorn) [reply parsers/src/x3d/x3d_parser.cpp:135]: Added exception in case if one of these is empty.

File: parsers/src/x3d/x3d_parser.cpp (hunk 2 above)
Comment (Sckorn) [reply parsers/src/x3d/x3d_parser.cpp]: Fixed

File: parsers/src/gltf/gltf_parser.cpp [hunk 5]
```diff
@@ -128,7 +129,7 @@ struct gltf_image {
 
 namespace nlohmann {
 template <>
-struct adl_serializer<gltf_pbr_met_rough> {
+struct R_PUBLIC_API adl_serializer<gltf_pbr_met_rough> {
```
[the remark this answers is not in the document]
Comment (Sckorn) [reply parsers/src/gltf/gltf_parser.cpp]: Removed

File: parsers/BUILD (hunk 1 above)
Comment (Sckorn) [reply parsers/BUILD]: Yup, removed everywhere.

File: parsers/src/gltf/gltf_parser.cpp [hunk 6]
```diff
@@ -150,7 +151,7 @@ struct adl_serializer<gltf_pbr_met_rough> {
 };
 
 template <>
-struct adl_serializer<gltf_material> {
+struct R_PUBLIC_API adl_serializer<gltf_material> {
```
Comment (radameon) [inline parsers/src/gltf/gltf_parser.cpp:154]: Still present

**Review by radameon — COMMENTED** [review]
SOme public prefixes are left.

File: parsers/src/gltf/gltf_parser.cpp (hunk 6 above)
Comment (Sckorn) [reply parsers/src/gltf/gltf_parser.cpp]: Removed

File: parsers/src/gltf/gltf_parser.cpp [hunk 7]
```diff
@@ -465,7 +466,7 @@ rcbe::geometry::Mesh::FacetStorage getValuesByAccessor(
 
 namespace rdmn::parse::gltf {
 
-std::vector<rcbe::core::CoreObject> parse(
+R_PUBLIC_API std::vector<rcbe::core::CoreObject> parse(
```
Comment (radameon) [inline parsers/src/gltf/gltf_parser.cpp:469]: Strictly speaking it's not required, but it's a minor thing, can be removed in the future.

**Review by radameon — APPROVED** [review]
Merge it, once CI is done.

**Pull request merged**